aten/src/ATen/native/mps/operations/LinearAlgebra.mm

```diff
@@ -780,6 +782,121 @@ static void linalg_lu_factor_out_mps_impl(const Tensor& A, bool pivot, Tensor& L
+  auto applyTRSMPSO = lib.getPipelineStateForFunc("applyTRSM");
+  auto applySYRKPSO = lib.getPipelineStateForFunc("applySYRK");
+
+  int64_t NB = std::min<int64_t>(32, N);
+  int64_t numBlocks = (N + NB - 1) / NB;
+
+  Tensor success = at::empty({B}, input.options().dtype(kInt)).fill_(1);
+  id<MTLBuffer> successBuffer = getMTLBufferStorage(success);
+
+  MTLSize threadGroupSize = MTLSizeMake(256, 1, 1);
+  id<MTLBuffer> outBuffer = getMTLBufferStorage(out);
+  id<MTLComputeCommandEncoder> computeEncoder = stream->commandEncoder();
+  [computeEncoder setBuffer:outBuffer offset:0 atIndex:0];
+
+  @autoreleasepool {
+    dispatch_sync_with_rethrow(stream->queue(), ^() {
+      for (int64_t k = 0; k < numBlocks; k++) {
+        uint32_t activeNB_k = (uint32_t)std::min(N - k * NB, (int64_t)NB);
+        [computeEncoder setComputePipelineState:factorDiagonalPSO];
+        std::array<uint32_t, 6> sizes = {static_cast<uint32_t>(N),
+                                         static_cast<uint32_t>(NB),
+                                         static_cast<uint32_t>(k),
+                                         static_cast<uint32_t>(activeNB_k),
+                                         static_cast<uint32_t>(B),
+                                         static_cast<uint32_t>(batch_stride)};
+        id<MTLBuffer> sizesBuffer = [device newBufferWithBytes:sizes.data()
+                                                        length:sizes.size() * sizeof(uint32_t)
+                                                       options:MTLResourceStorageModePrivate];
+        [computeEncoder setBuffer:sizesBuffer offset:0 atIndex:1];
+        [computeEncoder setBuffer:successBuffer offset:0 atIndex:2];
+        MTLSize gridSize = MTLSizeMake(B, 1, 1);
+        [computeEncoder dispatchThreadgroups:gridSize threadsPerThreadgroup:threadGroupSize];
+
+        // process all remaining blocks in this row/column in parallel
+        if (k < numBlocks - 1) {
+          int64_t startJ = k + 1;
+          int64_t nBlocksJ = (numBlocks - startJ);
+
+          if (nBlocksJ > 0) {
+            // TRSM for all blocks in parallel
+            uint32_t activeNB_k = (uint32_t)std::min(N - k * NB, (int64_t)NB);
+            std::array<uint32_t, 8> trsmSizes = {static_cast<uint32_t>(N),
+                                                 static_cast<uint32_t>(NB),
+                                                 static_cast<uint32_t>(k),
+                                                 static_cast<uint32_t>(startJ),
+                                                 static_cast<uint32_t>(nBlocksJ),
+                                                 static_cast<uint32_t>(activeNB_k),
+                                                 static_cast<uint32_t>(B),
+                                                 static_cast<uint32_t>(batch_stride)};
+
+            id<MTLBuffer> trsmSizesBuffer = [device newBufferWithBytes:trsmSizes.data()
```

Comment:
Hmm, this is not needed either, one can just pass array explciitly as an argument